src/pycsou/_dev/__init__.py

```diff
@@ -3,23 +3,37 @@
 import numpy as np
 
 import pycsou.abc.operator as pyco
+import pycsou.runtime as pycrt
 import pycsou.util as pycu
 import pycsou.util.ptype as pyct
 
 
-class SquaredL2Norm(pyco.DiffFunc):
-    def __init__(self, shape: pyct.ShapeOrDim = None):
-        super(SquaredL2Norm, self).__init__(shape=(1, None))
+class SquaredL2Norm(pyco.ProxDiffFunc):
+    # f: \bR^{M} -> \bR
+    #      x     -> \norm{x}{2}^{2}
+    def __init__(self, shape=None):
+        super().__init__(shape=(1, None))
+        self._lipschitz = np.inf
         self._diff_lipschitz = 2
 
-    def apply(self, arr: pyct.NDArray) -> pyct.Real:
+    @pycrt.enforce_precision(i="arr")
+    def apply(self, arr):
         xp = pycu.get_array_module(arr)
-        return xp.linalg.norm(arr, axis=-1, keepdims=True) ** 2
+        y = xp.linalg.norm(arr, axis=-1, keepdims=True)
+        y2 = xp.power(y, 2, dtype=arr.dtype)
+        return y2
 
-    def grad(self, arr: pyct.NDArray) -> pyct.NDArray:
+    @pycrt.enforce_precision(i="arr")
+    def grad(self, arr):
         return 2 * arr
 
-    def asloss(self, data: typ.Optional[pyct.NDArray] = None) -> pyco.DiffFunc:
+    @pycrt.enforce_precision(i=["arr", "tau"])
+    def prox(self, arr, tau):
+        y = arr / (2 * tau + 1)
+        return y
+
+    #    @pycrt.enforce_precision(i=["data"])
+    def asloss(self, data: typ.Optional[pyct.NDArray] = None) -> pyco.ProxFunc:
```

Comment:
This was my mistake, the base class uses `as_loss`, with **underscore**. We should use the same method name.